### ISSUE TITLE
fs-jetpack write causes "already exists" error

### ISSUE THREAD

**Issue by OwenBrotherwood** [szwacz/fs-jetpack#32]
On windows, I traced an error down to: 

`userDataDir.write(stateStoreFile, state, { atomic: true });
`
The error, after adding try/catch is

` saveStateError: EEXIST: file already exists, mkdir 'C:\Users\owen\AppData\Roaming'
`

---

I will investigate https://github.com/szwacz/fs-jetpack/blob/d3b02f4a3858d039653d03d96d469ae97a1bc82b/lib/write.js#L36

**Comment by szwacz** [comment]
Could you give me more insight when this bug happens?

**Comment by OwenBrotherwood** [comment]
Hi szacz,
I seem to have started the issue in the wrong repro, sorry, it should have been in electron-boilerplate.
And a bad first comment, let me try again.

In the saveState, I found I was getting an error: EEXIST: file already exists, mkdir 'C:\Users\owen\AppData\Roaming'
https://github.com/szwacz/electron-boilerplate/blob/6d9801d54474a89ccfbb82f70251edfd877550d5/src/helpers/window.js#L73

I have added more console.log's and I now believe that the product name is causing the problem, it includes an ":" and is probably a part of the path for the saveState

```
C:\Users\owen\Source\Repos\microbit-drill\microbit-drill-epxt>npm start

> @ start C:\Users\owen\Source\Repos\microbit-drill\microbit-drill-epxt
> gulp start

[21:46:02] Using gulpfile ~\Source\Repos\microbit-drill\microbit-drill-epxt\gulpfile.js
[21:46:02] Starting 'bundle'...
[21:46:02] Starting 'less'...
[21:46:02] Starting 'environment'...
[21:46:02] Finished 'environment' after 4.04 ms
[21:46:02] Starting 'watch'...
[21:46:02] Finished 'watch' after 11 ms
[21:46:02] Finished 'bundle' after 192 ms
[21:46:02] Finished 'less' after 203 ms
[21:46:02] Starting 'build'...
[21:46:02] Finished 'build' after 3.16 μs
[21:46:02] Starting 'start'...
[21:46:02] Finished 'start' after 5.02 ms

app.getPath('userData'):C:\Users\owen\AppData\Roaming\micro:bit drill ePTX (development)
saveState:stateStoreFile:window-state-main.json
saveState:Error: EEXIST: file already exists, mkdir 'C:\Users\owen\AppData\Roaming'

C:\Users\owen\Source\Repos\microbit-drill\microbit-drill-epxt>dir c:\Users\owen\AppData\Roaming\micro*.*
 Volume in drive C has no label.
 Volume Serial Number is 0EE0-492B

 Directory of c:\Users\owen\AppData\Roaming

15-10-2016  23:01    <DIR>          Microsoft FxCop
               0 File(s)              0 bytes
               1 Dir(s)  103.187.259.392 bytes free

C:\Users\owen\Source\Repos\microbit-drill\microbit-drill-epxt>

```

**Comment by szwacz** [comment]
ok, so closing

**Comment by OwenBrotherwood** [comment]
I investigate what is a lawfulname in "product name" which seems to be the directory electron-boilerplate helper/window.js bases it's state on and report back in electron-boilerplate if I find an issue (and hopefully come with a patch if necessary)
Br Owen
